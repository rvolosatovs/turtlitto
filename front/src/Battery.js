import React from "react";
import FontAwesomeIcon from "@fortawesome/react-fontawesome";
import faBatteryEmpty from "@fortawesome/fontawesome-free-solid/faBatteryEmpty";
import faBatteryQuarter from "@fortawesome/fontawesome-free-solid/faBatteryQuarter";
import faBatteryHalf from "@fortawesome/fontawesome-free-solid/faBatteryHalf";
import faBatteryThreeQuarters from "@fortawesome/fontawesome-free-solid/faBatteryThreeQuarters";
import faBatteryFull from "@fortawesome/fontawesome-free-solid/faBatteryFull";
import faAngleDown from "@fortawesome/fontawesome-free-solid/faAngleDown";

// Properties: percentage
const Battery = props => {
  const BatteryIcon = properties => {
    const percentage = properties.percentage;
    if (percentage < 10) {
      return (
        <FontAwesomeIcon icon={faBatteryEmpty} rotation={270} color="red" />
      );
    } else if (percentage >= 10 && percentage < 40) {
      return <FontAwesomeIcon icon={faBatteryQuarter} rotation={270} />;
    } else if (percentage >= 40 && percentage < 60) {
      return <FontAwesomeIcon icon={faBatteryHalf} rotation={270} />;
    } else if (percentage >= 60 && percentage < 90) {
      return <FontAwesomeIcon icon={faBatteryThreeQuarters} rotation={270} />;
    } else {
      return <FontAwesomeIcon icon={faBatteryFull} rotation={270} />;
    }
  };
  return (
    <div>
      {" "}
      <BatteryIcon icon={faBatteryFull} />
      {props.percentage}%
    </div>
  );
};

export default Battery;
