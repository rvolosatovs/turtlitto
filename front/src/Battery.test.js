import React from "react";
import Battery from "./Battery";
import renderer from "react-test-renderer";

test("Battery 100%", () => {
  const component = renderer.create(<Battery percentage={100} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Battery 75%", () => {
  const component = renderer.create(<Battery percentage={75} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Battery 50%", () => {
  const component = renderer.create(<Battery percentage={50} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Battery 25%", () => {
  const component = renderer.create(<Battery percentage={25} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Battery 0%", () => {
  const component = renderer.create(<Battery percentage={0} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
