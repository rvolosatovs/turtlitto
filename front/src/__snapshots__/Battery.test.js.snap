// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Battery 0% 1`] = `
<div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-battery-empty fa-w-20 fa-rotate-270 "
    color="red"
    data-icon="battery-empty"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M544 160v64h32v64h-32v64H64V160h480m16-64H48c-26.51 0-48 21.49-48 48v224c0 26.51 21.49 48 48 48h512c26.51 0 48-21.49 48-48v-16h8c13.255 0 24-10.745 24-24V184c0-13.255-10.745-24-24-24h-8v-16c0-26.51-21.49-48-48-48z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  0
  %
</div>
`;

exports[`Battery 25% 1`] = `
<div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-battery-quarter fa-w-20 fa-rotate-270 "
    data-icon="battery-quarter"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M544 160v64h32v64h-32v64H64V160h480m16-64H48c-26.51 0-48 21.49-48 48v224c0 26.51 21.49 48 48 48h512c26.51 0 48-21.49 48-48v-16h8c13.255 0 24-10.745 24-24V184c0-13.255-10.745-24-24-24h-8v-16c0-26.51-21.49-48-48-48zm-336 96H96v128h128V192z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  25
  %
</div>
`;

exports[`Battery 50% 1`] = `
<div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-battery-half fa-w-20 fa-rotate-270 "
    data-icon="battery-half"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M544 160v64h32v64h-32v64H64V160h480m16-64H48c-26.51 0-48 21.49-48 48v224c0 26.51 21.49 48 48 48h512c26.51 0 48-21.49 48-48v-16h8c13.255 0 24-10.745 24-24V184c0-13.255-10.745-24-24-24h-8v-16c0-26.51-21.49-48-48-48zm-240 96H96v128h224V192z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  50
  %
</div>
`;

exports[`Battery 75% 1`] = `
<div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-battery-three-quarters fa-w-20 fa-rotate-270 "
    data-icon="battery-three-quarters"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M544 160v64h32v64h-32v64H64V160h480m16-64H48c-26.51 0-48 21.49-48 48v224c0 26.51 21.49 48 48 48h512c26.51 0 48-21.49 48-48v-16h8c13.255 0 24-10.745 24-24V184c0-13.255-10.745-24-24-24h-8v-16c0-26.51-21.49-48-48-48zm-144 96H96v128h320V192z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  75
  %
</div>
`;

exports[`Battery 100% 1`] = `
<div>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-battery-full fa-w-20 fa-rotate-270 "
    data-icon="battery-full"
    data-prefix="fas"
    role="img"
    style={Object {}}
    viewBox="0 0 640 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M544 160v64h32v64h-32v64H64V160h480m16-64H48c-26.51 0-48 21.49-48 48v224c0 26.51 21.49 48 48 48h512c26.51 0 48-21.49 48-48v-16h8c13.255 0 24-10.745 24-24V184c0-13.255-10.745-24-24-24h-8v-16c0-26.51-21.49-48-48-48zm-48 96H96v128h416V192z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
  100
  %
</div>
`;
